include/GoogleSync/GoogleSync.php

```diff
@@ -46,1088 +53,249 @@
-     */
-    public function getAuthJson()
-    {
-        global $sugar_config;
-        $authJson_local = json_decode(base64_decode($sugar_config['google_auth_json']), true);
-        if (!$authJson_local) {
-        // The authconfig json string is invalid json
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Invalid AuthConfig JSON');
-            return false;
-        } elseif (!array_key_exists('web', $authJson_local)) {
-            // The authconfig is valid json, but the 'web' key is missing. This is not a valid authconfig.
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Invalid AuthConfig JSON');
-            return false;
-        } else {
-            return $authJson_local;
-        }
-    }
-
-    /**
-     * Add a user to the list of users to sync
-     *
-     * The user id is used as the key
-     *
-     * @param string $id : the SuiteCRM user id
-     * @param string $name : the SuiteCRM user name.
-     *  Not really used for anything other than reference.
-     *
-     * @return bool Success/Failure
-     */
-    public function addUser($id, $name)
-    {
-        if (array_key_exists($id, $this->users)) {
-            $GLOBALS['log']->warn(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . $id . ' already set');
-            return false;
-        } else {
-            $this->users[$id] = $name;
-            $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . $id . ' set to ' . $this->users[$id]);
-            return true;
-        }
-    }
-
-    /**
-     * Remove a user from the list of users to sync
-     *
-     * @param string $id : the SuiteCRM user id
-     *
-     * @return bool Success/Failure
-     */
-    public function delUser($id)
-    {
-        if (!array_key_exists($id, $this->users)) {
-            $GLOBALS['log']->warn(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . $id . ' missing');
-            return false;
-        } else {
-            unset($this->users[$id]);
-            return true;
-        }
-    }
+    protected $users = array();
 
     /**
-     * Creates and Sets the Google client in the object
-     *
-     * @param string $id : the SuiteCRM user id
-     *
-     * @return bool Success/Failure
+     * Gets the combined titles of a Meeting/Event pair for Logging
+     * 
+     * @param Meeting $meeting The CRM Meeting
+     * @param \Google_Service_Calendar_Event $event The Google Event
+     * 
+     * @return string The combined title
      */
-    public function setClient($id)
+    protected function getTitle(Meeting $meeting = null, Google_Service_Calendar_Event $event = null)
     {
-        if (!$gClient_local = $this->getClient($id)) {
-            return false;
-        } else {
-            $this->gClient = $gClient_local;
-            return true;
+        if (!$meeting) {
+            // TODO: should it be a user message?
```

Comment:
@gymad
I don't think this needs to be logged at all. It would be completely normal for this function to only get one event or the other. If both are missing, it sets it to "UNNAMED RECORD". No error.

Reply:
Fixed in temp branch